src/client/auth/ticketZKProofMulti.ts

```diff
@@ -0,0 +1,148 @@
+import { AbstractAuthentication, AuthenticationMethodMulti, AuthenticationResult } from './abstractAuthentication'
+import { OffChainTokenConfig, MultiTokenInterface } from '../interface'
+import { OutletAction, Messaging } from '../messaging'
+import { Authenticator } from '@tokenscript/attestation'
+import { SignedUNChallenge } from './signedUNChallenge'
+import { UNInterface } from './util/UN'
+import { LocalOutlet } from '../../outlet/localOutlet'
+import { OutletIssuerInterface } from '../../outlet'
+import { createIssuerHashArray } from '../../utils'
+import { shouldUseRedirectMode } from '../../utils/support/getBrowserData'
+import { EasZkProof } from '@tokenscript/attestation/dist/eas/EasZkProof'
+import { DEFAULT_EAS_SCHEMA, TokenType } from '../../outlet/ticketStorage'
+import { EAS_RPC_CONFIG } from '../../core/eas'
+
+export class TicketZKProofMulti extends AbstractAuthentication implements AuthenticationMethodMulti {
+	TYPE = 'ticketZKProofMulti'
+
+	private messaging = new Messaging()
+
+	// NOTE: A limitation at this time is that the user can only authenticate tokens
+	// that use the same issuer origin, email and it's underlying config (see: firstCollectionFound via getTokenProofMulti).
+
+	async getTokenProofMulti(userTokens: MultiTokenInterface[], request: any): Promise<AuthenticationResult> {
+		const firstCollectionFound = Object.keys(userTokens)[0]
+		const tokenCollectionConfig = userTokens[firstCollectionFound].requestTokens[0]
+		const issuerCollectionConfig = userTokens[firstCollectionFound].issuerConfig
+		let redirectMode: false | string = this.getRedirectModeSetting(tokenCollectionConfig, request)
```

Comment:
@nicktaras better disable option to force redirect. it depends on browser only. for testing we can use Brave or Safari and Chrome. simplier code = less errors and easier to update

Reply:
I'll split this into a new Jira ticket. For us to remove this, but also for the examples that uses this as a means of testing + any of the BDD/E2E testing suite that uses this trigger to simulate the tests.

Reply:
https://smarttokenlabs.atlassian.net/browse/TIKAT-293 - Moved to a seperate task for us to take on. Thanks for the suggestion to look at doing this.

Reply:
we can easy change user-agent in testing , no need to pass specific param. @nicktaras 
thank for a task!